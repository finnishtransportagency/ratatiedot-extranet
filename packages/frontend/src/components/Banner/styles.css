.banner-alert {
  transform: scale(1);
  animation: pulse 2s 3;
}

@keyframes pulse {
  0% {
    transform: scale(1);
  }
  50% {
    transform: scale(1.1);
  }
  100% {
    transform: scale(1);
  }
}
