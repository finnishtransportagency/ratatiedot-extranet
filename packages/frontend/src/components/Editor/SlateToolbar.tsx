import styled from '@emotion/styled';
import { Box, Divider, IconButton, Paper, ToggleButton, ToggleButtonGroup } from '@mui/material';
import FormatBoldIcon from '@mui/icons-material/FormatBold';
import FormatItalicIcon from '@mui/icons-material/FormatItalic';
import FormatUnderlinedIcon from '@mui/icons-material/FormatUnderlined';
import FormatListBulletedIcon from '@mui/icons-material/FormatListBulleted';
import FormatListNumberedIcon from '@mui/icons-material/FormatListNumbered';
import { Slate } from 'slate-react';

import RedoIcon from '@mui/icons-material/Redo';
import UndoIcon from '@mui/icons-material/Undo';
import CheckIcon from '@mui/icons-material/Check';
import CloseIcon from '../../assets/icons/Close.svg';
import LinkIcon from '../../assets/icons/Link.svg';
import PaletteIcon from '../../assets/icons/Palette.svg';
import { insertLink, isBlockActive, isMarkActive, toggleBlock, toggleMark } from '../../utils/slateEditorUtil';
import { Colors } from '../../constants/Colors';
import { ElementType, FontFormatType } from '../../utils/types';
import { NotificationTypes } from './NotificationTypes';
import { useContext, useEffect, useState } from 'react';
import { AppBarContext } from '../../contexts/AppBarContext';
import { EditorContext } from '../../contexts/EditorContext';
import { useTranslation } from 'react-i18next';
import { EditorColorPicker } from './Popup/EditorColorPicker';
import { ContentTypes } from './ContentTypes';
import { FontSizeDropdown } from './Dropdown/FontSizeDropdown';
import { ButtonWrapper } from './ConfirmationAppBar';
import { useUpdatePageContents } from '../../hooks/mutations/UpdateCategoryPageContent';
import { getRouterName } from '../../utils/helpers';
import { useLocation, useMatch } from 'react-router-dom';
import { toast } from 'react-toastify';
import { isEqual } from 'lodash';
import { HistoryEditor } from 'slate-history';
import { useUpdateNoticePageContents } from '../../hooks/mutations/UpdateNoticePageContent';

type MarkButtonProps = { editor: any; format: FontFormatType; icon: any };

type BlockButtonProps = {
  editor: any;
  format: ElementType;
  icon: any;
};

const MarkButton = ({ editor, format, icon }: MarkButtonProps) => {
  const { t } = useTranslation(['common']);

  return (
    <ToggleButton
      aria-label={t(`common:font.${format}`)}
      value={format}
      selected={isMarkActive(editor, format)}
      onMouseDown={(event: React.MouseEvent<HTMLElement>) => {
        event.preventDefault();
        toggleMark(editor, format);
      }}
    >
      {icon}
    </ToggleButton>
  );
};

const BlockButton = ({ editor, format, icon }: BlockButtonProps) => {
  const { t } = useTranslation(['common']);

  return (
    <ToggleButton
      aria-label={t(`common:element.${format}`)}
      value={format}
      selected={isBlockActive(editor, format)}
      onMouseDown={(event: React.MouseEvent<HTMLElement>) => {
        event.preventDefault();
        toggleBlock(editor, format);
      }}
    >
      {icon}
    </ToggleButton>
  );
};

export const SlateToolbar = () => {
  const { t } = useTranslation(['common']);
  const { closeToolbarHandler, closeToolbarWithoutSaveHandler } = useContext(AppBarContext);
  const { editor, value, valueReset, noticeFields } = useContext(EditorContext);
  const [isColorOpened, setIsColorOpened] = useState(false);
  const [initialValue, setInitialValue] = useState([]);
  const { pathname } = useLocation();
  const categoryName = pathname.split('/').at(-1) || '';
  const noticeRoute = useMatch('/ajankohtaista/:id');

  const mutatePageContents = useUpdatePageContents(getRouterName(categoryName));
  const mutateNoticePageContents = useUpdateNoticePageContents(categoryName);

  const { error } = mutatePageContents;

  useEffect(() => {
    setInitialValue(value);
  }, []);
  const handleInsertLink = () => {
    const url = prompt(t('common:edit.enter_url'));
    if (!url) return;
    insertLink(editor, url);
  };

  const toggleColorPicker = () => setIsColorOpened(!isColorOpened);

  const handleSave = () => {
    if (noticeRoute) {
      console.log('Save on noticeRoute');
      mutateNoticePageContents.mutate(
        { value, noticeFields },
        {
          onSuccess: () => {
            toast(t('common:edit.saved_success'), { type: 'success' });
            setInitialValue(value);
          },
          onError: () => {
            toast(error ? error.message : t('common:edit.saved_failure'), { type: 'error' });
          },
        },
      );
    } else {
      mutatePageContents.mutate(value, {
        onSuccess: () => {
          toast(t('common:edit.saved_success'), { type: 'success' });
          setInitialValue(value);
        },
        onError: () => {
          toast(error ? error.message : t('common:edit.saved_failure'), { type: 'error' });
        },
      });
    }
  };

  const handleClose = () => {
    if (isEqual(initialValue, value)) {
      closeToolbarWithoutSaveHandler();
    } else {
      closeToolbarHandler();
    }
  };

  return (
    <Slate editor={editor} initialValue={value}>
      <ToolbarPaperWrapper elevation={2} aria-label={t('common:edit.toolbar')}>
        <FontSizeDropdown />
        <DividerWrapper orientation="vertical" variant="middle" flexItem />
        <IconButton onClick={() => HistoryEditor.undo(editor)}>
          <UndoIcon fontSize="small" />
        </IconButton>
        <IconButton onClick={() => HistoryEditor.redo(editor)}>
          <RedoIcon fontSize="small" />
        </IconButton>
        <DividerWrapper orientation="vertical" variant="middle" flexItem />
        <ToggleButtonGroupWrapper size="small">
          {MarkButton({ editor, format: FontFormatType.BOLD, icon: <FormatBoldIcon fontSize="small" /> })}
          {MarkButton({ editor, format: FontFormatType.ITALIC, icon: <FormatItalicIcon fontSize="small" /> })}
          {MarkButton({ editor, format: FontFormatType.UNDERLINED, icon: <FormatUnderlinedIcon fontSize="small" /> })}
          <Box
            aria-label={t('common:edit.insert_link')}
            component="img"
            sx={{ cursor: 'pointer', width: '25px', padding: '7px' }}
            src={LinkIcon}
            alt="link"
            onClick={handleInsertLink}
          />
          {BlockButton({
            editor,
            format: ElementType.NUMBERED_LIST,
            icon: <FormatListNumberedIcon fontSize="small" />,
          })}
          {BlockButton({ editor, format: ElementType.BULLET_LIST, icon: <FormatListBulletedIcon fontSize="small" /> })}
        </ToggleButtonGroupWrapper>
        <DividerWrapper orientation="vertical" variant="middle" flexItem />
        <Box
          aria-label={t('common:edit.color')}
          component="img"
          sx={{ cursor: 'pointer', width: '25px' }}
          src={PaletteIcon}
          alt="color"
          onClick={toggleColorPicker}
        />
        {isColorOpened && (
          <ToggleButtonGroupWrapper size="small">
            <EditorColorPicker onClose={toggleColorPicker} />
          </ToggleButtonGroupWrapper>
        )}
        <DividerWrapper orientation="vertical" variant="middle" flexItem />
        <NotificationTypes />
        <DividerWrapper orientation="vertical" variant="middle" flexItem />
        <ContentTypes />
        <DividerWrapper orientation="vertical" variant="middle" flexItem />
        <ButtonWrapper onClick={valueReset}>{t('common:action.reject')}</ButtonWrapper>
        <ButtonWrapper variant="contained" onClick={handleSave}>
          <CheckIcon fontSize="small" />
          {t('common:action.save')}
        </ButtonWrapper>
        <Box
          aria-label={t('common:action.close')}
          component="img"
          sx={{ cursor: 'pointer', marginLeft: 'auto' }}
          src={CloseIcon}
          alt="close"
          onClick={handleClose}
        />
      </ToolbarPaperWrapper>
    </Slate>
  );
};

const ToolbarPaperWrapper = styled(Paper)(({ theme }) => ({
  display: 'flex',
  borderRadius: 0,
  boxShadow: 'none',
  flexWrap: 'wrap',
  backgroundColor: Colors.aliceblue,
  borderBottom: `1px dashed ${Colors.darkblue}`,
  padding: '16px 15px',
}));

const ToggleButtonGroupWrapper = styled(ToggleButtonGroup)(() => ({
  '.MuiToggleButton-root': { border: 'none' },
}));

const DividerWrapper = styled(Divider)(({ theme }) => ({
  margin: '0 27px',
  [theme.breakpoints.down('desktop')]: {
    display: 'none',
  },
}));
