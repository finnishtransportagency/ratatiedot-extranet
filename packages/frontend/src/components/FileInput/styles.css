.file-input {
  box-sizing: border-box;
  width: 100%;
  padding: 10px 8px;
  border: 1px solid #333333;
  border-radius: 4px;
  font-family: 'Exo2-Regular';
  font-size: 16px;
}
