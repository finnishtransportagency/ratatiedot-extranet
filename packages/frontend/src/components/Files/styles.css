.additional-form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  box-sizing: border-box;
  width: 100%;
}

.expand-button {
  transition: transform 0.2s ease-in-out;
}

.expand-button.active {
  transform: rotate(180deg);
}
