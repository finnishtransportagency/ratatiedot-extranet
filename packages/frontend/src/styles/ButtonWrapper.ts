import { Button } from '@mui/material';
import styled from '@emotion/styled';

export const ButtonWrapper = styled(Button)(({ theme }) => {
  return {
    marginTop: theme.spacing(4),
    padding: '9px',
    borderRadius: '100px',
    [theme.breakpoints.down('tablet')]: {
      width: '100%',
    },
    [theme.breakpoints.up('tablet')]: {
      width: '220px',
      height: '42px',
    },
  };
}) as typeof Button;
