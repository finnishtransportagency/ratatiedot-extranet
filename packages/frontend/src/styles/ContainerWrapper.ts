import { Box } from '@mui/material';
import styled from '@emotion/styled';

export const ContainerWrapper = styled(Box)(({ theme }) => ({
  [theme.breakpoints.down('tablet')]: {
    margin: '15px 16px',
  },
  [theme.breakpoints.up('tablet')]: {
    margin: theme.spacing(4),
  },
  [theme.breakpoints.up('desktop')]: {
    margin: `${theme.spacing(5)} auto`,
    width: '65%',
  },
})) as typeof Box;
