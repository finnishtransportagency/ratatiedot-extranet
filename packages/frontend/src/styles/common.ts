import { Box, Button, Typography } from '@mui/material';
import styled from '@emotion/styled';

export const ContainerWrapper = styled(Box)(({ theme }) => ({
  [theme.breakpoints.down('tablet')]: {
    margin: '15px 16px',
  },
  [theme.breakpoints.up('tablet')]: {
    margin: theme.spacing(4),
  },
  [theme.breakpoints.up('desktop')]: {
    margin: `${theme.spacing(5)} auto`,
    width: '65%',
  },
})) as typeof Box;

export const ProtectedContainerWrapper = styled(Box)(({ theme }) => ({
  [theme.breakpoints.only('mobile')]: {
    margin: '16px 15px',
  },
  [theme.breakpoints.only('tablet')]: {
    margin: theme.spacing(4),
    marginTop: 0,
  },
  [theme.breakpoints.up('desktop')]: {
    margin: theme.spacing(5),
  },
}));

export const PageTitleWrapper = styled(Typography)(({ theme }) => {
  return {
    fontWeight: 700,
    fontSize: '23px',
    marginBottom: '25px',
    [theme.breakpoints.only('desktop')]: {
      fontSize: '28px',
      marginBottom: '35px',
    },
  };
});

export const ButtonWrapper = styled(Button)(({ theme }) => {
  return {
    marginTop: theme.spacing(4),
    padding: '9px',
    borderRadius: '100px',
    [theme.breakpoints.down('tablet')]: {
      width: '100%',
    },
    [theme.breakpoints.up('tablet')]: {
      width: '220px',
      height: '42px',
    },
  };
}) as typeof Button;
