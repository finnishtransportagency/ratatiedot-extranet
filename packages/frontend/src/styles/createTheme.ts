import { createTheme } from '@mui/material';

import { DefaultFonts } from '../constants/Fonts';

import '../assets/fonts/Exo_2/Exo2-Regular.ttf';
import '../assets/fonts/Exo_2/Exo2-ExtraBold.ttf';
import '../assets/fonts/Exo_2/Exo2-Bold.ttf';
import '../assets/fonts/Exo_2/Exo2-LightItalic.ttf';
import { Colors } from '../constants/Colors';
import { Viewports } from '../constants/Viewports';

// Default theme object: https://mui.com/material-ui/customization/default-theme/#main-content
export const theme = createTheme({
  spacing: 8, // default 8px
  breakpoints: {
    values: {
      mobile: Viewports.MOBILE,
      tablet: Viewports.TABLET,
      desktop: Viewports.DESKTOP,
    },
  },
  palette: {
    primary: {
      main: Colors.darkblue,
      light: Colors.lightblue,
      dark: Colors.darkblue,
    },
    secondary: {
      main: Colors.extrablack,
      light: Colors.darkgrey,
    },
    error: {
      main: Colors.darkred,
      light: Colors.lightred,
    },
    warning: {
      main: Colors.yellow,
    },
    success: {
      main: Colors.darkgreen,
      light: Colors.lightgreen,
    },
    info: {
      main: Colors.midblue,
      light: Colors.aliceblue,
    },
    grey: {
      50: Colors.lightgrey,
      100: Colors.lightgrey,
      200: Colors.midgrey,
      300: Colors.midgrey,
      500: Colors.darkgrey,
      600: Colors.darkgrey,
    },
    background: {
      default: Colors.white,
      paper: Colors.white,
    },
    transparent: {
      main: Colors.white,
    },
  },
  // Typography reference: https://www.figma.com/file/RowcUf6WZvVTqkChr6IRG6/Extranet?node-id=1%3A379
  typography: {
    fontFamily: ['Exo2-Regular', ...DefaultFonts].join(','),
    fontSize: 16,
    subtitle1: {
      fontFamily: ['Exo2-Bold', ...DefaultFonts].join(','),
      fontSize: 23,
    },
    subtitle2: {
      fontFamily: ['Exo2-Bold', ...DefaultFonts].join(','),
      fontSize: 16,
    },
    h4: {
      fontFamily: ['Exo2-Bold', ...DefaultFonts].join(','),
      fontSize: 16,
      lineHeight: '24px',
    },
    body1: {
      fontFamily: ['Exo2-Regular', ...DefaultFonts].join(','),
      fontSize: 16,
    },
    body2: {
      fontFamily: ['Exo2-Regular', ...DefaultFonts].join(','),
      fontSize: 18,
    },
    button: {
      fontFamily: ['Exo2-Regular', ...DefaultFonts].join(','),
      fontSize: 20,
      color: 'white',
      textTransform: 'uppercase',
      textAlign: 'center',
    },
  },
});
