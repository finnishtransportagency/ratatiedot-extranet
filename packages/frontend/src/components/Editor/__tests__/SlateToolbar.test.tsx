import { ThemeProvider } from '@mui/material';
import { cleanup, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { theme } from '../../../styles/createTheme';
import { SlateToolbar } from '../SlateToolbar';

describe('SlateToolbar component', () => {
  let component = null as any;
  beforeEach(() => {
    component = (
      <ThemeProvider theme={theme}>
        <SlateToolbar />
      </ThemeProvider>
    );
  });

  afterEach(() => {
    component = null;
    cleanup();
  });

  test('SlateToolbar should render properly', () => {
    render(component);
  });

  test('SlateToolbar should match snapshot', () => {
    const slateToolbar = renderer.create(component).toJSON();
    expect(slateToolbar).toMatchSnapshot();
  });

  test('SlateToolbar should have a list of control buttons', () => {
    render(component);
    expect(screen.getByRole('button', { name: /paragraph-two/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /paragraph-one/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /heading-two/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /bold/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /italic/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /underlined/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /numbered-list/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /bulleted-list/i })).toBeInTheDocument();
    expect(screen.getByRole('img', { name: /insert-link/i })).toBeInTheDocument();
    expect(screen.getByRole('img', { name: /info/i })).toBeInTheDocument();
    expect(screen.getByRole('img', { name: /warning/i })).toBeInTheDocument();
    expect(screen.getByRole('img', { name: /error/i })).toBeInTheDocument();
    expect(screen.getByRole('img', { name: /check/i })).toBeInTheDocument();
    expect(screen.getByRole('img', { name: /delete/i })).toBeInTheDocument();
    expect(screen.getByRole('img', { name: /close/i })).toBeInTheDocument();
  });
});
