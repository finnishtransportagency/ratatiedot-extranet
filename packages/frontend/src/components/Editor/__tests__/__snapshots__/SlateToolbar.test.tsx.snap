// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlateToolbar component SlateToolbar should match snapshot 1`] = `
<div
  aria-label="työkalupalkki"
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-1qnc7df-MuiPaper-root"
>
  <div
    aria-label="Valitse fontin koko"
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-n0xngz-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Leipäteksti
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      tabIndex={-1}
      value="paragraph-two"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1rejtkv-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
  <div
    className="MuiToggleButtonGroup-root css-4hldsh-MuiToggleButtonGroup-root"
    role="group"
  >
    <button
      aria-label="lihavoitu"
      aria-pressed={null}
      className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-aaezgw-MuiButtonBase-root-MuiToggleButton-root"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
      value="bold"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-y3v3gx-MuiSvgIcon-root"
        data-testid="FormatBoldIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.6 10.79c.97-.67 1.65-1.77 1.65-2.79 0-2.26-1.75-4-4-4H7v14h7.04c2.09 0 3.71-1.7 3.71-3.79 0-1.52-.86-2.82-2.15-3.42zM10 6.5h3c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5h-3v-3zm3.5 9H10v-3h3.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5z"
        />
      </svg>
    </button>
    <button
      aria-label="kursivoitu"
      aria-pressed={null}
      className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-aaezgw-MuiButtonBase-root-MuiToggleButton-root"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
      value="italic"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-y3v3gx-MuiSvgIcon-root"
        data-testid="FormatItalicIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 4v3h2.21l-3.42 8H6v3h8v-3h-2.21l3.42-8H18V4z"
        />
      </svg>
    </button>
    <button
      aria-label="alleviivattu"
      aria-pressed={null}
      className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-aaezgw-MuiButtonBase-root-MuiToggleButton-root"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
      value="underlined"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-y3v3gx-MuiSvgIcon-root"
        data-testid="FormatUnderlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17c3.31 0 6-2.69 6-6V3h-2.5v8c0 1.93-1.57 3.5-3.5 3.5S8.5 12.93 8.5 11V3H6v8c0 3.31 2.69 6 6 6zm-7 2v2h14v-2H5z"
        />
      </svg>
    </button>
    <img
      alt="link"
      aria-label="Lisää linkki"
      className="MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiBox-root css-1xst3u4"
      disabled={false}
      fullWidth={false}
      onChange={[Function]}
      onClick={[Function]}
      selected={false}
      size="small"
      src="Link.svg"
    />
    <button
      aria-label="numeroitu lista"
      aria-pressed={false}
      className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-aaezgw-MuiButtonBase-root-MuiToggleButton-root"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
      value="numbered-list"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-y3v3gx-MuiSvgIcon-root"
        data-testid="FormatListNumberedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M2 17h2v.5H3v1h1v.5H2v1h3v-4H2v1zm1-9h1V4H2v1h1v3zm-1 3h1.8L2 13.1v.9h3v-1H3.2L5 10.9V10H2v1zm5-6v2h14V5H7zm0 14h14v-2H7v2zm0-6h14v-2H7v2z"
        />
      </svg>
    </button>
    <button
      aria-label="numeroimaton lista"
      aria-pressed={false}
      className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-aaezgw-MuiButtonBase-root-MuiToggleButton-root"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
      value="bulleted-list"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-y3v3gx-MuiSvgIcon-root"
        data-testid="FormatListBulletedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M4 10.5c-.83 0-1.5.67-1.5 1.5s.67 1.5 1.5 1.5 1.5-.67 1.5-1.5-.67-1.5-1.5-1.5zm0-6c-.83 0-1.5.67-1.5 1.5S3.17 7.5 4 7.5 5.5 6.83 5.5 6 4.83 4.5 4 4.5zm0 12c-.83 0-1.5.68-1.5 1.5s.68 1.5 1.5 1.5 1.5-.68 1.5-1.5-.67-1.5-1.5-1.5zM7 19h14v-2H7v2zm0-6h14v-2H7v2zm0-8v2h14V5H7z"
        />
      </svg>
    </button>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-middle MuiDivider-vertical MuiDivider-flexItem css-1dpfv1n-MuiDivider-root"
  />
  <img
    alt="color"
    aria-label="väri"
    className="MuiBox-root css-us34y6"
    onClick={[Function]}
    src="Palette.svg"
  />
  <hr
    className="MuiDivider-root MuiDivider-middle MuiDivider-vertical MuiDivider-flexItem css-1dpfv1n-MuiDivider-root"
  />
  <div
    className="css-k008qs"
  >
    <img
      alt="info"
      aria-label="info"
      className="MuiBox-root css-4g6ai3"
      onClick={[Function]}
      src="Add_info.svg"
    />
    <img
      alt="warning"
      aria-label="varoitus"
      className="MuiBox-root css-4g6ai3"
      onClick={[Function]}
      src="Add_varoitus.svg"
    />
    <img
      alt="error"
      aria-label="virhe"
      className="MuiBox-root css-4g6ai3"
      onClick={[Function]}
      src="Add_virhe.svg"
    />
    <img
      alt="check"
      aria-label="oikein-merkki"
      className="MuiBox-root css-4g6ai3"
      onClick={[Function]}
      src="Add_vahvistus.svg"
    />
  </div>
  <hr
    className="MuiDivider-root MuiDivider-middle MuiDivider-vertical MuiDivider-flexItem css-1dpfv1n-MuiDivider-root"
  />
  <div
    className="css-k008qs"
  >
    <img
      alt="card"
      aria-label="kortti"
      className="MuiBox-root css-4g6ai3"
      onClick={[Function]}
      src="Add_kortti.svg"
    />
    <img
      alt="text"
      aria-label="teksti"
      className="MuiBox-root css-4g6ai3"
      onClick={[Function]}
      src="Add_teksti.svg"
    />
  </div>
  <hr
    className="MuiDivider-root MuiDivider-middle MuiDivider-vertical MuiDivider-flexItem css-1dpfv1n-MuiDivider-root"
  />
  <img
    alt="close"
    aria-label="Sulje"
    className="MuiBox-root css-1yceefw"
    onClick={[Function]}
    src="Close.svg"
  />
</div>
`;
